stdlib/src/collections/counter.mojo

```diff
@@ -76,3 +82,400 @@ struct Counter[V: KeyElement](
+    # Comparison operators
+    # ===-------------------------------------------------------------------===#
+
+    fn __eq__(self, other: Self) -> Bool:
+        """Check if all counts agree. Missing counts are treated as zero.
+
+        Args:
+            other: The other Counter to compare to.
+
+        Returns:
+            True if the two Counters are equal, False otherwise.
+        """
+
+        @parameter
+        @always_inline
+        fn is_eq(keys: _DictKeyIter[V, Int, _]) -> Bool:
+            for e_ref in keys:
+                var e = e_ref[]
+                if self.get(e, 0) != other.get(e, 0):
+                    return False
+            return True
+
+        return is_eq(self.keys()) and is_eq(other.keys())
+
+    fn __ne__(self, other: Self) -> Bool:
+        """Check if all counts disagree. Missing counts are treated as zero.
+
+        Args:
+            other: The other Counter to compare to.
+
+        Returns:
+            True if the two Counters are not equal, False otherwise.
+        """
+        return not self == other
+
+    fn __le__(self, other: Self) -> Bool:
+        """Check if all counts are less than or equal to the other Counter.
+
+        Args:
+            other: The other Counter to compare to.
+
+        Returns:
+            True if all counts are less than or equal to the other Counter, False otherwise.
+        """
+
+        @parameter
+        @always_inline
+        fn is_le(keys: _DictKeyIter[V, Int, _]) -> Bool:
+            for e_ref in keys:
+                var e = e_ref[]
+                if self.get(e, 0) > other.get(e, 0):
+                    return False
+            return True
+
+        return is_le(self.keys())
+
+    fn __lt__(self, other: Self) -> Bool:
+        """Check if all counts are less than in the other Counter.
+
+        Args:
+            other: The other Counter to compare to.
+
+        Returns:
+            True if all counts are less than in the other Counter, False otherwise.
+        """
+        return self <= other and self != other
+
+    fn __gt__(self, other: Self) -> Bool:
+        """Check if all counts are greater than in the other Counter.
+
+        Args:
+            other: The other Counter to compare to.
+
+        Returns:
+            True if all counts are greater than in the other Counter, False otherwise.
+        """
+        return other < self
+
+    fn __ge__(self, other: Self) -> Bool:
+        """Check if all counts are greater than or equal to the other Counter.
+
+        Args:
+            other: The other Counter to compare to.
+
+        Returns:
+            True if all counts are greater than or equal to the other Counter, False otherwise.
+        """
+        return other <= self
+
+    # ===-------------------------------------------------------------------===#
+    # Binary operators
+    # ===-------------------------------------------------------------------===#
+
+    fn __add__(self, other: Self) raises -> Self:
```

Comment:
Why does this raise?

Reply:
Thanks for the catch! Done here: https://github.com/msaelices/mojo/commit/d5f109f1f4dbb07f8de94366da6344473264a7fa

Reply:
Also implemented the useful `Counter.__contains__()` method here: https://github.com/msaelices/mojo/commit/0f22d2800ea813ec2742b1d9bc935b325219de61